File: src/main/java/easy/PalindromeNumber.java
package easy;

public class PalindromeNumber {
  public Boolean solve(int x) {
    return this.isNumberPalindromic(x);
  }

  public Boolean isNumberPalindromic(int x) {
    if (x < 0 || (x % 10 == 0 && x != 0)) {
      return false;
    }

    String number = Integer.toString(x);
    int left = 0, right = number.length() - 1;
    while (left < right) {
      if (number.charAt(left) != number.charAt(right)) {
        return false;
      }
      right--;
      left++;
    }
    return true;
  }
}

File: src/easy/palindrome-number.ts
export const palindromeNumber = (x: number): boolean => {
    if (x < 0 || (x % 10 == 0 && x != 0)) return false;
    const num = x.toString();
    let right = num.length - 1;
    for (const left of num) {
        if (left !== num[right]) return false;
        right -= 1;
    }
    return true;
};
